src/index.js

```diff
@@ -0,0 +1,324 @@
+        this.connected = false;
+        this.authenticated = false;
+        this.linked = false;
+        this.initialised = false;
+        this.socket = null;
+        this.appName = null;
+        this.otp = null;
+        this.openRequests = [];
+        this.origin = null;
+        this.socket = null;
+    }
+    reset() {
+        this.connected = false;
+        this.authenticated = false;
+        this.linked = false;        
+        this.socket = null;        
+        this.otp = null;
+        this.openRequests = [];        
+        this.socket = null;
+    }
+    async init(app) {
+        if (this.initialised === false) {
+            this.appName = app;
+            this.origin = app;
+            if (typeof location !== 'undefined') {
+                if (location.hasOwnProperty('hostname') && location.hostname.length && location.hostname !== 'localhost') {
+                    this.origin = location.hostname;
+                }
+            }
+        }
+        this.detected = browser();
+        this.keyhash = CryptoJS.SHA256(this.detected.name + ' ' + this.origin + ' ' + this.appName).toString();
+        let appstore;
+        appstore = await BeetClientDB.apps.where("keyhash").equals(this.keyhash).toArray();
+        this.initialised = true;
+        return appstore;
+    }
+    ping() {
+        let ping;
+        return new Promise(async (resolve, reject) => {
+            try {
+                ping = new WebSocket(this.host);
+                ping.onopen = function (evt) {
+                    ping.send('{ "type" : "version"}');
+                }
+                ping.onmessage = function (evt) {
+                    let msg = JSON.parse(evt.data);
+                    if (msg.type == "version") {
+                        resolve(msg.result);
+                    } else {
+                        reject(false);
+                    }
+                    ping.close();
+                }
+            } catch (e) {
+                reject(false);
+            }
+        });
+    }
+    isInstalled() {
+        return new Promise(resolve => {
+            setTimeout(() => {
+                resolve(false);
+            }, 3000);
+            this.ping().then(found => {
+                if (found) resolve(found);
+            });
+
+        })
+    }
+    async get_id() {
+        return Math.round(Math.random() * 100000000 + 1);
+    }
+    async fetch_ids() {
+        
+        let app = await BeetClientDB.apps.where("apphash").equals(this.identity.apphash).first();        
+        let id = app.next_id;
+        let new_id = await this.next_id();
+        let next_hash = await CryptoJS.SHA256('' + new_id).toString();
+        return {
+            id: id,
+            next_hash: next_hash.toString()
+        };
+    }
+    async next_id() {
+        if (this.connected && this.authenticated && this.linked) {
+            let next_id = Math.round(Math.random() * 100000000 + 1);
```

Comment:
Not using get_id() ?

Reply:
good point....I refactored that part a few times so I didn't notice